overlord/snapstate/snapstate.go

```diff
@@ -179,6 +157,36 @@ func TryPath(s *state.State, name, path string, flags Flags) (*state.TaskSet, er
 	return InstallPath(s, name, path, "", flags)
 }
 
+// Install returns a set of tasks for installing snap.
+// Note that the state must be locked by the caller.
+func Install(s *state.State, name, channel string, userID int, flags Flags) (*state.TaskSet, error) {
+	var snapst SnapState
+	err := Get(s, name, &snapst)
+	if err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+	if snapst.CurrentSideInfo() != nil {
+		return nil, fmt.Errorf("snap %q already installed", name)
+	}
+
+	snapInfo, err := snapInfo(s, name, channel, userID, flags)
+	if err != nil {
+		return nil, err
+	}
+	snapst.Candidate = &snapInfo.SideInfo
```

Comment:
the issue is that if we fail to create the Change that candidate should not be there ideally